src/pages/home/HeaderView.js

```diff
@@ -288,25 +294,22 @@ HeaderView.displayName = 'HeaderView';
 HeaderView.defaultProps = defaultProps;
 
 export default memo(
-    compose(
-        withWindowDimensions,
-        withLocalize,
-        withOnyx({
-            guideCalendarLink: {
-                key: ONYXKEYS.ACCOUNT,
-                selector: (account) => (account && account.guideCalendarLink) || null,
-                initialValue: null,
-            },
-            parentReport: {
-                key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT}${report.parentReportID || report.reportID}`,
-                selector: reportWithoutHasDraftSelector,
-            },
-            session: {
-                key: ONYXKEYS.SESSION,
-            },
-            policies: {
-                key: ONYXKEYS.COLLECTION.POLICY,
-            },
-        }),
-    )(HeaderView),
+    withOnyx({
+        guideCalendarLink: {
+            key: ONYXKEYS.ACCOUNT,
+            selector: (account) => (account && account.guideCalendarLink) || null,
+            initialValue: null,
+        },
+        parentReport: {
+            key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT}${report.parentReportID || report.reportID}`,
+            selector: reportWithoutHasDraftSelector,
+        },
+        session: {
+            key: ONYXKEYS.SESSION,
+        },
+        policy: {
+            key: ({report}) => `${ONYXKEYS.COLLECTION.POLICY}${report ? report.policyID : '0'}`,
+            selector: (policy) => _.pick(policy, ['name', 'avatar', 'pendingAction']),
```

Comment:
Question: doesn't this break the memoization (this is wrapped by `memo()`) if this selector is always returning a new instance of the sub-object picked?

Reply:
> Does this break something or cause regression?

No that I know of, I just saw the code and got curious. I think it would be good to know if it makes the `memo` completely useless. If it does, it may have been better to just return the complete policy object 🤷

Reply:
I'm assuming that the implementation of `memo` is not doing a deep comparison, it is just doing `===`, but maybe I'm wrong about that.

Reply:
> I think that fine because we are also using only memo in other places without callback comparison.

It could also be wrong there... I guess I'll have to test it :P

Reply:
I was able to test and I confirmed that I'm wrong, the `memo` is working fine.


In case you are interested, I tested by adding this code in ReportScreen to force it to re-render every second

```
    const [forceRerender, setForceRerender] = useState(0);
    useEffect(() => {
        setTimeout(()=>{
            setForceRerender(forceRerender + 1);
        }, 1000);
    }, [forceRerender])
    console.log('ReportScreen rerender', forceRerender);
```

and also added a console log in `HeaderView`. I can see the `ReportScreen rerender` every second, but `HeaderView rerender` doesn't appear. If I remove the `memo`, the `HeaderView rerender` starts appearing every time `ReportScreen` rerenders.